packages/flutter/lib/src/painting/_network_image_io.dart

```diff
@@ -107,6 +155,62 @@ class NetworkImage extends image_provider.ImageProvider<image_provider.NetworkIm
     }
   }
 
+  static Future<Isolate> _pendingLoader;
+  static List<Function> _pendingLoaderUsers;
+  static SendPort _requestPort;
+
+  Future<ui.Codec> _loadAsyncOnDesignatedIsolate(
```

Comment:
I'd prefer that this function is split into two functions, one idempotent function that's concerned with lazy-initializing the isolate, and a second function that sends a download request to the isolate. The latter would call the former.